### PR TITLE
A buyer can update their cart and also their total items count will b…

### PR DESCRIPTION
**Pull request by bahati10**
Certainly! Here's the PR template using the specified format:

---

### What does this PR do?
This PR implements the functionality to handle updates to the buyer's cart, including storing the updated cart in the buyer's session and calculating the new cart total.

### Description of Task to be completed?
- Handle `post` requests to update the cart via the cart ID.
- Store the updated cart in the buyer's session.
- Retrieve product details (**name**, **price**, **image**) from the database for each item in the updated cart.
- Calculate the new cart total by summing the prices of the items in the updated cart.
- Send an updated cart confirmation message to the frontend to be displayed to the buyer.

### How should this be manually tested?
1. Send a `POST` request to update the cart with a valid cart ID and updated quantities.
2. Verify that the updated cart is stored in the buyer's session.
3. Check that the product details for each item in the updated cart are correctly retrieved from the database.
4. Confirm that the new cart total is accurately calculated.
5. Ensure an updated cart confirmation message is sent to the frontend.

#### N/A

---

This template should help in creating a comprehensive and clear pull request for updating the cart functionality.

### CODE REVIEW

File: src/services/cart.services.ts [hunk 1]
```diff
@@ -90,3 +93,54 @@ export const addToCart = async (quantity: number, product: Product, user: UserAt
...
+    }
+
+    await CartItem.destroy({ where: { id: cartItemId } });
+
+    return cartItem;
+  } catch (error: any) {
+    console.error('Error deleting cart item:', error.message);
+    throw new Error(error.message);
+  }
+};
```
Comment (kirengamartial) [inline src/services/cart.services.ts:146]: remember to add new lines for prettier

**Review by kirengamartial — APPROVED** [review]
the code looks good and clean but  check for the changes i told you to make

File: migrations/20240603131254-add-totalitems-to-cart-items.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up (queryInterface, Sequelize) {
+    /**
+     * Add altering commands here.
+     *
```
Comment (kirengamartial) [inline migrations/20240603131254-add-totalitems-to-cart-items.js]: remove this file since it has only comments and it is not doing anything to our project

**Review by kirengamartial — APPROVED** [review]

File: migrations/20240603131254-add-totalitems-to-cart-items.js (hunk 2 above)
Comment (bahati10) [reply migrations/20240603131254-add-totalitems-to-cart-items.js]: Done! appreciations for pointing that out

**Review by kirengamartial — DISMISSED** [review]
solve the gitguardian check

**Review by Ishimwe7 — APPROVED** [review]
Good job your codes are clear, Bahati I think your branch is ready to be merged

**Review by Balinda21 — APPROVED** [review]
Well done!

**Review by DeliceLydia — APPROVED** [review]

**Pull request merged**